thunderfish/fakefish.py

```diff
@@ -0,0 +1,173 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def generate_wavefish(freq, samplerate, time_len=20., harmonics=3):
+    """
+
+    :param freq: (float). Frequency of the fish in Hz.
+    :param samplerate: (float). Sampling rate in Hz.
+    :param time_len: (float). Length of the recording in sec.
+    :param harmonics: (int). Number of harmonics of freq to be included.
+
+    :return: (array). Data of a wavefish with given frequency.
+    """
+    time = np.arange(0, time_len, 1./samplerate)
+
+    data = np.sin(2 * np.pi * time * freq)
+    for har in range(harmonics):
+        if har > 1:
+            data += (np.sin(2*np.pi*time*(har*freq))) * 0.1*har
+
+    # Insert some noise
+    data += 0.05 * np.random.randn(len(data))
+
+    return data
+
+
+def generate_pulsefish(freq, samplerate, time_len=20., noise_fac=0.1, wave_cut=0.6,
+                       pk_len=0.001, pk_std=0.1, peak_fac=2.,
+                       tr_len=0.001, tr_std=0.12, tr_fac=1.2):
+    """
+
+    :param freq: (float). Frequency of the fish in Hz.
+    :param samplerate: (float). Sampling Rate in Hz
+    :param time_len: (float). Length of the recording in sec.
+    :param noise_fac: (float). Factor by which random gaussian distributed noise is inserted.
+    :param wave_cut: (float). Ideally between 0.5 and 1. marks the percentage at which the distribution is cut.
+    :param pk_len: (float). length of the positive part of the pulse.
+    :param pk_std: (float). std of the positive part of the pulse.
+    :param peak_fac: (float). Factor for regulating the positive part of the pulse.
+    :param tr_len: (float). length of the negative part of the pulse.
+    :param tr_std: (float). std of the negative part of the pulse.
+    :param tr_fac: (float). Factor for regulating the negative part of the pulse.
```

Comment:
you do not need pk_len and th_len. You only need to know the distance between the pulses.